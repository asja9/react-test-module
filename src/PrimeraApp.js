/*Para definir un componente debe ser el archivo en Mayus*/

/*Para poder usar JSX nesecito importar react */
 import React, { Fragment } from "react";
 /**Para poder realizar la validacion */
 import PropTypes from 'prop-types';


 /*Definicion de la aplicacion
   paso un parametro en la aplicacion y aplico deestructuracion*/
const PrimeraApp = ( {mensaje, subTitulo}) =>{


    const saludo = 'Hola perro';
    //console.log(mensaje);

    //Return de los resultados, en parentesis si son varias lineas.
    //El retorno html deberia ser envuelto en un div padre
    /*JSX dse reqiere React exportado*/
    /*
    return (
            <div>
             <h1>Hola mundo</h1>
             <h2>Hola React</h2>
            </div>
          );
    */
    //Si no quiero agregar div, se resuelve con fragment
    /*
    return (
        <Fragment>
         <h1>Hola mundo</h1>
         <h2>Hola React</h2>
        </Fragment>
      );
    */
    
    /*Forma resumida de envolver*/
      return (
        <>
         <h1>{saludo}</h1>
         <h2>{mensaje}</h2>
         <h3 id="sub">{subTitulo}</h3>
        </>
      );
}

/** Validacion de los parametros de entrada */
PrimeraApp.propTypes = {
    mensaje: PropTypes.string.isRequired
}

//Parametros por default de mu modulo
PrimeraApp.defaultProps = {
  subTitulo: 'Soy un canijo subtitulo'
}


export default PrimeraApp;