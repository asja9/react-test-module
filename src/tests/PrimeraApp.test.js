//Debe ser un import o no va manejar bien el error
import { render } from '@testing-library/react';

//no es necesario, para que muestre las ayudas
import '@testing-library/jest-dom'

import PrimeraApp from '../PrimeraApp';
import React from 'react';

//Libreria para pruebas de modulos
import { shallow } from 'enzyme';

//Revisar Dom para la prueba
import '@testing-library/jest-dom/extend-expect';

describe('Pruebas en <PrimeraApp></PrimeraApp>', () =>{

    test('Debe mostrar el mensaje "Hola, soy Goku', () =>{
        const mensaje = "Hola, soy Goku";

        const {getByText} = render(<PrimeraApp mensaje = {mensaje} />);
        
        console.log(getByText);
        expect( getByText( mensaje ) ).toBeInTheDocument();
        //Esperar que exista saludo en el documento

    });
    
    //Prueba usando Enzyme
    test("Debe de mostrar <PrimeraApp></PrimeraApp> correctamente con Enzyme", () =>{
        
        const mensaje = "Hola, soy Goku";
        const wrapper = shallow(<PrimeraApp mensaje = {mensaje} />);
   
        expect(wrapper).toMatchSnapshot()
   
    });
    
});

/*comando para ejecutat laa pruebas*/
    