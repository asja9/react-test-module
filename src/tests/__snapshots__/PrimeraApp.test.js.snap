// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp></PrimeraApp> Debe de mostrar <PrimeraApp></PrimeraApp> correctamente con Enzyme 1`] = `
<Fragment>
  <h1>
    Hola perro
  </h1>
  <h2>
    Hola, soy Goku
  </h2>
  <h3
    id="sub"
  >
    Soy un canijo subtitulo
  </h3>
</Fragment>
`;
